### PR TITLE
CPP-5241 Migrate github action from build-wrapper-output to compilation-database

### PR DESCRIPTION
**Pull request by enzo-pellegrini**

### CODE REVIEW

File: scripts/configure_paths.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-if [[ ${ARCH} != "X64" ]]; then 
+if [[ ${ARCH} != "X64" && ! ( ${OS} == "macOS" && ${ARCH} == "ARM64" ) ]]; then
```
Comment (michael-jabbour-sonarsource) [inline scripts/configure_paths.sh:3]: I would at least move this change to a separate PR. FYI, there is a ticket for it: https://sonarsource.atlassian.net/browse/SQCPPGHA-2

File: README.md [hunk 2]
```diff
@@ -59,15 +59,15 @@ jobs:
...
+      # here goes your compilation wrapped with build-wrapper; See https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/languages/c-family/#using-build-wrapper for more information
       # build-preparation steps
       # build-wrapper-linux-x86-64 --out-dir ${{ env.BUILD_WRAPPER_OUT_DIR }} build-command
     - name: Run sonar-scanner
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
-      run: sonar-scanner --define sonar.cfamily.build-wrapper-output="${{ env.BUILD_WRAPPER_OUT_DIR }}" #Consult https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/scanners/sonarscanner/ for more information and options
+      run: sonar-scanner --define sonar.cfamily.compile-commands="${{ env.BUILD_WRAPPER_OUT_DIR }}/compile_commands.json" #Consult https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/scanners/sonarscanner/ for more information and options
```
Comment (michael-jabbour-sonarsource) [inline README.md:70]: For SQ, I would rather keep both examples for now. For example, A snippet before SQ 10.6 that uses build-wrapper-output, and a snippet after 10.6 that uses the compile-commands property.

Comment (enzo-pellegrini) [reply README.md:70]: How would you mention both in this readme? would put the run in two separate commands and use comments to say what to do based on the version of sonarqube, or put two completely separate yaml examples?

Comment (michael-jabbour-sonarsource) [reply README.md:70]: I didn't take a close look, I'll let you decide on the clearest presentation format. My initial feeling is that duplicating the yaml with changing only one line doesn't help with clarity, so I am leaning towards comments inside the YAML.

File: README.md [hunk 3]
```diff
@@ -59,15 +59,15 @@ jobs:
...
 ```
```
Comment (enzo-pellegrini) [inline README.md:72]: What if we mentioned the modification to the command in this section, after the yaml? The yaml is supposed to be an example anyway, "looks like this".
We can say something like **If you are using LTS** please use `sonar.cfamily.build-wrapper-output = ...` instead of the current one?

Comment (michael-jabbour-sonarsource) [reply README.md:72]: Let's add a separate paragraph as you suggested..

**Review by michael-jabbour-sonarsource — APPROVED** [review]
LGTM, I think that we can merge this PR around the time SQ 10.6 is released to avoid confusion.

@enzo-pellegrini Could you please amend the commit and include the ticket number. We can then request the final review from the AE squad...
